### PR TITLE
change io to use spectra from deep by default

### PR DESCRIPTION
**Pull request by Waelthus**
This is to allow using the deep spectra by default in DESI SV mode, there's 2 flags for picca_deltas to allow changing to all or allow to use the previous behaviour, i.e. selecting only spectra from specific nights.

### CODE REVIEW

File: py/picca/io.py [hunk 1]
```diff
@@ -1069,19 +1076,39 @@ def read_from_minisv_desi(in_dir, catalog, pk1d=None):
...
+            files_in = glob.glob(os.path.join(in_dir, "**/all/**/coadd-*.fits"),
+                         recursive=True)
+        else:
+            files_in = glob.glob(os.path.join(in_dir, "**/deep/**/coadd-*.fits"),
+                         recursive=True)
+        petal_tile = [
+            f"{entry['PETAL_LOC']}-{entry['TILEID']}"
+            for entry in catalog
+        ]
+        petal_tile_unique = np.unique(petal_tile)
```
Comment (andreufont) [inline py/picca/io.py:1104]: Why do we need this "unique" call here? Do we have expect more than one file for this combination of petal and tile?

**Review by andreufont — COMMENTED** [review]
The code looks good, but I haven't tested the branch. @2199433g - could you take a look at it?

**Review by 2199433g — APPROVED** [review]
Picca_deltas works for me with --use-all!

**Comment by Waelthus** [comment]
ok, seems to do its job with single nights or the new default to use deep stacks as well

File: py/picca/io.py (hunk 1 above)
Comment (Waelthus) [reply py/picca/io.py:1104]: yes, this goes over the objects in the cat not over the files, only files that have objects in the cat will then be read from disk at all.

**Comment by andreufont** [comment]
We can discuss this at the telecon today, and merge right after? 

I'm still confused about why we need to use petal_tile_unique instead of directly petal_tile, but I guess that's ok. What would happen if we didn't add that line? Maybe add a comment explaining this to the code?

**Comment by Waelthus** [comment]
I think that without the line files might appear multiple times in the tile list and thus might be read-in multiple times. But I didn't test for this specifically...

**Comment by andreufont** [comment]
Even if we are only reading coadd files under /deep/ ?

If this happened, it might be better to write an error message and crash, unless we understand why this would happen.

**Comment by Waelthus** [comment]
it happens due to the 2-level for-loop after that loops over all files and for each file over all cat entries. I now let it break when the first match is found so that each file that has at least one match in the cat should appear exactly once in the filelist. The unique is thus not strictly needed, but should speed up (but this gain is probably unimportant) the double loop that follows right after (which is ugly, but was the easiest I came up with when writing this piece last year).

**Comment by andreufont** [comment]
I see. We can keep it as it is for now, but it would be good if we found a nice way to do this long-term. We should be using this function quite a lot in the next few months.

Thanks for getting this started, by the way, it has been very useful.

**Comment by Waelthus** [comment]
Yes, I didn't want to be overly sophisticated here, also because we're likely replacing those routines soon.

**Pull request merged**